File: source/algorithm/NQ014/NQ014_improved.py
# NQ014 凤凰花开吃杨梅
def calc(n):
    res=1  # 第n天剩下1个
    for i in range(n-1,0,-1):
        res=(res+1)*2  # 逆推前一天的数量
    return res

def main():
    import sys
    for line in sys.stdin:
        line=line.strip()
        if not line:continue
        x=int(line)
        print(calc(x))  # 计算并输出

if __name__=="__main__":main()

File: 兴趣班入门百练讲义/NQ100/NQ014/NQ014_improved.cpp
// NQ014 凤凰花开吃杨梅
#include <iostream>
using namespace std;
int calc(int n){
    int res=1;  // 第n天剩下1个
    for(int i=n-1;i>0;--i){
        res=(res+1)*2;  // 逆推前一天的数量
    }
    return res;
}
int main(){
    int x;
    while(cin>>x){
        cout<<calc(x)<<'\n';  // 计算并输出
    }
    return 0;
}